File: 3321-type-of-triangle/3321-type-of-triangle.py
class Solution:
    def triangleType(self, nums: List[int]) -> str:
        nums.sort()
        if nums[2]>=nums[0]+nums[1]: return "none"
        if nums[0]==nums[2]: return "equilateral"
        if nums[0]==nums[1] or nums[1]==nums[2]: return "isosceles"
        return "scalene"

File: 3321-type-of-triangle/3321-type-of-triangle.c
#pragma GCC optimize("O3", "unroll-loops")
inline void swap(int* x, int* y){int tmp=*x; *x=*y; *y=tmp;}

char* triangleType(int* nums, int numsSize) {
    for(int i=0; i<2; i++){
        if (nums[2]<nums[i]) swap(&nums[i], &nums[2]);
        else if (nums[1]<nums[i]) swap(&nums[i], &nums[1]);
    }
    if (nums[2]>=nums[0]+nums[1]) return "none";
    if (nums[0]==nums[2]) return "equilateral";
    if (nums[0]==nums[1] || nums[1]==nums[2]) return "isosceles";
    return "scalene";
}